src/ConcordiumNetSdk/Transactions/ITransactionSigner.cs

```diff
@@ -1,31 +1,33 @@
-﻿using ConcordiumNetSdk.SignKey;
-using Index = ConcordiumNetSdk.Types.Index;
+﻿using System.Collections.Immutable;
+using ConcordiumNetSdk.SignKey;
 
 namespace ConcordiumNetSdk.Transactions;
 
 /// <summary>
-/// A transaction signer is a map from the index of the credential to another map from the key index to the actual signer.
-/// The credential index is relative to the account address, and the indices should be distinct.
-/// The key index is relative to the credential.
-/// The maximum length of the list is 255, and the minimum length is 1.
+/// A transaction signer is a map from indices of credentials to maps from key indices to signers.
+/// A credential index is relative to the account address and a key index is relative to the credential.
+/// The number of credentials and keys per credential should be at least 1 and at most 255.
```

Comment:
I think this should be explained a bit. There are too many indices here.

There should be a bit of an explanation of account keys
- account can have one or more credential
- a credential has one or more (up to 255) keys.

Each credential has a threshold for how many signatures are needed to authorize. And the account has a threshold on how many credentials need to sign to authorize transactions from the account.